tests/org.eclipse.ui.workbench.texteditor.tests/src/org/eclipse/ui/workbench/texteditor/tests/DialogAccess.java

```diff
@@ -0,0 +1,253 @@
+import org.eclipse.ui.internal.findandreplace.SearchOptions;
+
+class DialogAccess implements IFindReplaceUIAccess {
+
+	FindReplaceLogic findReplaceLogic;
+
+	Combo findCombo;
+
+	Combo replaceCombo;
+
+	Button forwardRadioButton;
+
+	Button globalRadioButton;
+
+	Button searchInRangeRadioButton;
+
+	Button caseCheckBox;
+
+	Button wrapCheckBox;
+
+	Button wholeWordCheckBox;
+
+	Button incrementalCheckBox;
+
+	Button regExCheckBox;
+
+	Button findButton;
+
+	Button replaceButton;
+
+	Button replaceAllButton;
+
+	private Supplier<Shell> shellRetriever;
+
+	private Runnable closeOperation;
+
+	private FindReplaceUITest<DialogAccess> parentTest;
+
+	Accessor dialogAccessor;
+
+	DialogAccess(FindReplaceUITest<DialogAccess> parent, Accessor findReplaceDialogAccessor) {
+		dialogAccessor= findReplaceDialogAccessor;
+		parentTest= parent;
+		findReplaceLogic= (FindReplaceLogic) findReplaceDialogAccessor.get("findReplaceLogic");
+		findCombo= (Combo) findReplaceDialogAccessor.get("fFindField");
+		replaceCombo= (Combo) findReplaceDialogAccessor.get("fReplaceField");
+		forwardRadioButton= (Button) findReplaceDialogAccessor.get("fForwardRadioButton");
+		globalRadioButton= (Button) findReplaceDialogAccessor.get("fGlobalRadioButton");
+		searchInRangeRadioButton= (Button) findReplaceDialogAccessor.get("fSelectedRangeRadioButton");
+		caseCheckBox= (Button) findReplaceDialogAccessor.get("fCaseCheckBox");
+		wrapCheckBox= (Button) findReplaceDialogAccessor.get("fWrapCheckBox");
+		wholeWordCheckBox= (Button) findReplaceDialogAccessor.get("fWholeWordCheckBox");
+		incrementalCheckBox= (Button) findReplaceDialogAccessor.get("fIncrementalCheckBox");
+		regExCheckBox= (Button) findReplaceDialogAccessor.get("fIsRegExCheckBox");
+		shellRetriever= () -> ((Shell) findReplaceDialogAccessor.get("fActiveShell"));
+		closeOperation= () -> findReplaceDialogAccessor.invoke("close", null);
+		findButton= (Button) findReplaceDialogAccessor.get("fFindNextButton");
+		replaceButton= (Button) findReplaceDialogAccessor.get("fReplaceSelectionButton");
+		replaceAllButton= (Button) findReplaceDialogAccessor.get("fReplaceAllButton");
+	}
+
+	void restoreInitialConfiguration() {
+		findCombo.setText("");
+		select(SearchOptions.FORWARD);
+		select(SearchOptions.GLOBAL);
+		select(SearchOptions.WRAP);
+		unselect(SearchOptions.INCREMENTAL);
+		unselect(SearchOptions.REGEX);
+		unselect(SearchOptions.CASE_SENSITIVE);
+		unselect(SearchOptions.WHOLE_WORD);
+	}
+
+	@Override
+	public Button getButtonForSearchOption(SearchOptions option) {
+		switch (option) {
+			case CASE_SENSITIVE:
+				return caseCheckBox;
+			case FORWARD:
+				return forwardRadioButton;
+			case GLOBAL:
+				return globalRadioButton;
+			case INCREMENTAL:
+				return incrementalCheckBox;
+			case REGEX:
+				return regExCheckBox;
+			case WHOLE_WORD:
+				return wholeWordCheckBox;
+			case WRAP:
+				return wrapCheckBox;
+			default:
+				return null;
```

Comment:
![grafik](https://github.com/eclipse-platform/eclipse.platform.ui/assets/16443184/90dfbc4b-6f94-49ca-ba3d-82443d495914)

This is the output of the Eclipse-Linter. Since the default-case does not really add complexity l and the Linter says so, I am in favor of keeping the default-case. What do you think?